airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -142,3 +141,18 @@ def set_autocommit(self, conn, autocommit: Any) -> None:
 
     def get_autocommit(self, conn):
         return getattr(conn, 'autocommit_mode', False)
+
+    def run(self, sql, autocommit=False, parameters=None):
+        """
+        Snowflake-connector doesn't allow natively the execution of multiple SQL statements in the same
+        call. So for allowing to pass files or strings with several queries this method is coded,
+        that relies on run from DBApiHook
+        """
+        if isinstance(sql, str):
+            with closing(self.get_conn()) as conn:
+                if self.supports_autocommit:
+                    self.set_autocommit(conn, autocommit)
+
+                    conn.execute_string(sql, parameters)
+        else:
+            super().run(sql, autocommit, parameters)
```

Comment:
Can you add a test for this @JavierLopezT

Reply:
Hi @kaxil . I think I have to test that depending on whether sql is a list or a string, it calls conn.execute_string or super().run, but I have no idea how to do that. Could you help me, please? Thank you very much in advance

Reply:
Hey @JavierLopezT -- Have a look at https://github.com/apache/airflow/blob/master/tests/providers/exasol/hooks/test_exasol.py for some examples.

Let me know if you need help though

Reply:
@JavierLopezT what you want to do is use mock.patch to verify that in the `run` method, the right underlying method is called given the right input.  google for basic examples of mock patch.

in this case you would want to mock the `get_conn` method so that its `return_value` is a mock object.  then after `run` finishes, you can verify that the mock object was called with method `execute_string` and your value for `sql`

there are lots of examples in the repo just try to find something similar